### PR TITLE
Chakra install enhancements

### PR DESCRIPTION
**Pull request by danmih-ixia**
- build and install with pip install from local source or directly from github
- use pyproject.toml to describe the project attibutes and dependecies in a declarative way
- install dependencies when installing the package
- upgrade to protobuf 4.x
- build proto file with setuptools-grpc instead of protoc; this removes the need of installing protobuf-compiler
- fix imports from internal modules to be able to be called from anywhere, not only from the cloned repo root.
- et_generator: add comm_size attribute to one_comm_send/recv_node

### CODE REVIEW

**Comment by srinivas212** [comment]
Thanks for enabling this feature, @danmih-ixia. Can you fix the merge conflict. I will also try this out and add more comments if any.

**Comment by JoongunPark** [comment]
Hello, I met below error while building the code. Could you check this please?

```
python3 -m chakra.et_generator.et_generator\
    --num_npus 64\
    --num_dims 1
```

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/a./anaconda3/lib/python3.11/site-packages/chakra/et_generator/et_generator.py", line 6, in <module>
    from chakra.et_def.et_def_pb2 import (
ImportError: cannot import name 'BOOL' from 'chakra.et_def.et_def_pb2' (/Users/a./anaconda3/lib/python3.11/site-packages/chakra/et_def/et_def_pb2.py)

**Comment by JoongunPark** [comment]
I also figured out that et_generator works without Taekyung's two commits(a2b45c10fed84bae0515e706b74063dcf99d57d0, c236cfaeeff459cd508fb266b6e0e2b63dd44e87) in the middle.

File: INSTALL.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+$ pip install .
+
+# Install latest from github
+$ pip install https://github.com/mlcommons/chakra/archive/refs/heads/main.zip
+
+# Install specific revision from github
+$ pip install https://github.com/mlcommons/chakra/archive/ae7c671db702eb1384015bb2618dc753eed787f2.zip
+
+# Install as dependency from requirements.txt
+$ cat requirements.txt
```
Comment (TaekyungHeo) [inline INSTALL.md:15]: @danmih-ixia  I cannot find requirements.txt. It seems that you have removed it. Could you please explain the reason?

File: INSTALL.md [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+chakra@https://github.com/mlcommons/chakra/archive/ae7c671db702eb1384015bb2618dc753eed787f2.zip
```
Comment (TaekyungHeo) [inline INSTALL.md:16]: I don't understand this output of `cat requirements.txt`

File: INSTALL.md (hunk 1 above)
Comment (danmih-ixia) [reply INSTALL.md:15]: Requirements.txt is not required anymore as all of the requirements are mentioned as dependencies in pyproject.toml and installed as dependencies when doing pip install. There is no need to run separately the pip install -r requirements.txt after that anymore.

File: INSTALL.md (hunk 2 above)
Comment (danmih-ixia) [reply INSTALL.md:16]: it shows what to put in the requirements.txt file in an external project to bring in chakra as a dependency.

**Comment by TaekyungHeo** [comment]
@danmih-ixia , there is a failing build test. Could you please check?

File: INSTALL.md (hunk 2 above)
Comment (danmih-ixia) [reply INSTALL.md]: I removed this section.

**Comment by danmih-ixia** [comment]
Fixed the failing build test.

File: setup.py [hunk 3]
```diff
@@ -1,38 +1,9 @@
...
-        author="MLCommons",
-        author_email="chakra@mlcommons.org",
-        url="https://github.com/mlcommons/chakra",
-        packages=packages,
-        package_dir=package_dir_map
-    )
 
-if __name__ == "__main__":
-    main()
+setup(cmdclass={'build': build_grpc})
```
Comment (TaekyungHeo) [inline setup.py:9]:
```suggestion
setup(
    version='0.0.4', 
    cmdclass={'build': build_grpc},
)
```

Comment (srinivas212) [reply setup.py:9]: @TaekyungHeo thanks for the suggestion.

@danmih-ixia  I found different install commands used different versions. the above fix should take care of it. Rest looks good to me. Please fix and I will merge this important PR. Apologies for the delay in reviewing.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+[build-system]
+requires = ["setuptools", "setuptools-grpc"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "chakra"
+requires-python = ">=3.7"
+version = "0.0.3"
```
Comment (srinivas212) [inline pyproject.toml:8]: This should be 0.0.4. 

cc: @TaekyungHeo

Comment (danmih-ixia) [reply pyproject.toml:8]: Correct. The version is specified in pyproject.toml. I'll update to 0.0.4, I did not update after the new version was merged.

**Review by srinivas212 — APPROVED** [review]
Tested locally. LGTM.

**Pull request merged**